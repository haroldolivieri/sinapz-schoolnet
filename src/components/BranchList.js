import React from 'react'
import SinapzCard from './SinapzCard'

const BranchList = ({ branches }) => {
  if(!branches || !branches.length){
    return <div>Essa rede n√£o tem filiais</div>
  }
  return branches.map(branch => {
    return(
      <SinapzCard branch={branch} key={branch.id} />
    )
  })
}

export default BranchList
