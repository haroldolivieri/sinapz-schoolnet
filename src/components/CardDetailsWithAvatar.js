import React from 'react'
import Card from '@material-ui/core/Card'
import CardHeader from '@material-ui/core/CardHeader'
import Avatar from '@material-ui/core/Avatar'
import '../css/CardDetailsWithAvatar.css'
import PropTypes from 'prop-types'
import { extractInitials } from '../util/handlerWithNames'

export const CardDetailsWithAvatar = ({branch}) => {
  return (
    <Card key={branch.id} className="card-container">
      <div className="avatar-item">
        <CardHeader avatar={
          <Avatar>{ extractInitials(branch.name) }</Avatar>
        }
        />
      </div>
      <div className="content-item">
        <span className="item-card-text-name">{branch.name}</span>
        <span className="item-card-text-sub">{`Respons√°vel: ${branch.inCharge}`}</span>
      </div>
    </Card>
  )
}

CardDetailsWithAvatar.propTypes = {
  branch: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    inCharge: PropTypes.string
  })
}

CardDetailsWithAvatar.defaultProps = {
  branch: {
    id: "Unknown",
    name: "Unknown",
    inCharge: "Unknown"
  }
}

export default CardDetailsWithAvatar