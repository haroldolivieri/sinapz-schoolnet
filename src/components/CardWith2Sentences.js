import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import '../css/CardWith2Sentences.css'

export const CardWith2Sentences = ({
                            FirstSentence = [<b className='highlight'>Hello</b>],
                            SecondSentence = [<b>]World!</b>],
}) => (
        <Card class="card">
            <CardContent  class="card_content">
                { FirstSentence }
                { SecondSentence }
            </CardContent>
        </Card>
);

CardWith2Sentences.propTypes = {
    FirstSentence: PropTypes.string,
    SecondSentence: PropTypes.string,
};