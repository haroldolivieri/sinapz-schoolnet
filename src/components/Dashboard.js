import React from 'react';
import Grid from "@material-ui/core/Grid";
import { DashboardCard, DashboardPaper } from '.'
import PropTypes from 'prop-types'

export const DashboardIndice = ({
    quantidade = 0,
    tipo = 'Unknown',
}) => (
    <Grid item md={4}>
        <DashboardCard
            // backgroundColor="#00f"
            quantidade = { quantidade }
            tipo = { tipo }
        />
    </Grid>
)
DashboardIndice.propTypes = {
    quantidade: PropTypes.number,
    tipo: PropTypes.string,
}

export const DashboardAvaliacao = ({
    tipo = 'Unknown',
    disciplina = 'Unknown',
    grupo = 'Unknown',
    tempoRestante = 'Unknown'
}) => (
    <DashboardPaper
        tipo = { tipo }
        disciplina = { disciplina }
        grupo = { grupo}
        tempoRestante = { tempoRestante }
    />
)
DashboardAvaliacao.propTypes = {
    tipo: PropTypes.string,
    disciplina: PropTypes.string,
    grupo: PropTypes.string,
    tempoRestante: PropTypes.string,
}

const Dashboard = () => (
    <div>
        <div>
            <Grid container spacing={24}>
            {/* Start DashboardIndices Temp */}
                <Grid item md={4}>
                    <DashboardCard
                        // backgroundColor="#00f"
                        quantidade = "5"
                        tipo = "Filiais"
                    />
                </Grid>
                <Grid item md={4}>
                    <DashboardCard
                        // backgroundColor="#00f"
                        quantidade = "40"
                        tipo = "Professores"
                    />
                </Grid>
                <Grid item md={4}>
                    <DashboardCard
                        // backgroundColor="#00f"
                        quantidade = "200"
                        tipo = "Alunos"
                    />
                </Grid>
            {/* End Temp */}
            {/* {DashboardIndices.map(item => (
                    <DashboardIndice
                        quantidade={item.quantidade}
                        tipo={item.tipo}
                    />
                ))} */}
            </Grid>
        </div>
        <div>
            <h1>Próximas Avaliações em 24h </h1>
            {/* Start DashboardAvaliacoes Temp */}
            <DashboardPaper
                tipo={"Prova"}
                disciplina={ "Geografia" }
                grupo = { "Ensino Médio"}
                tempoRestante={ "30m" }
            />
            <br/>
            <DashboardPaper
                tipo={"Teste"}
                disciplina={ "Matematica" }
                grupo = { "Ensino Fundamental"}
                tempoRestante={ "1h" }
            />
            <br/>
            <DashboardPaper
                tipo={"Prova"}
                disciplina={ "Geografia" }
                grupo = { "Ensino Médio"}
                tempoRestante={ "2h" }
            />
            <br/>
            <DashboardPaper
                tipo={"Prova"}
                disciplina={ "Geografia" }
                grupo = { "Ensino Médio"}
                tempoRestante={ "30m" }
            />
            <br/>
            <DashboardPaper
                tipo={"Teste"}
                disciplina={ "Matematica" }
                grupo = { "Ensino Fundamental"}
                tempoRestante={ "1h" }
            />
            <br/>
            <DashboardPaper
                tipo={"Prova"}
                disciplina={ "Geografia" }
                grupo = { "Ensino Médio"}
                tempoRestante={ "2h" }
            />
            <br/>
            {/* End DashboardAvaliacoes Temp */}
            {/* {DashboardAvaliacoes.map(item => (
                    <DashboardAvaliacao
                        tipo={item.tipo}
                        disciplina={item.disciplina}
                        grupo={item.grupo}
                        tempoRestante={item.tempoRestante}
                    />
                ))} */}
        </div>
    </div>
)

export default Dashboard
