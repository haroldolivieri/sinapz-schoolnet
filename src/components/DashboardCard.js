import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import '../css/DashboardCard.css'


function SimpleCard(props) {

    const {
        quantidade,
        tipo,
    } = props;

    return (
        <div class="card" >
            <Card>
                <CardContent  class="card_content">
                    <Typography variant="h1" component="h1" >
                        <b>{ quantidade }</b>
                    </Typography>
                    <Typography variant="h2" component="h2">
                        <b>{ tipo }</b>
                    </Typography>
                </CardContent>
            </Card>
        </div>
    );
}

SimpleCard.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default 
    SimpleCard;
