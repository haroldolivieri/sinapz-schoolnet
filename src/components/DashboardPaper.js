import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import '../css/DashboardPaper.css'

const styles = theme => ({
    root: {
        ...theme.mixins.gutters(),
        paddingTop: theme.spacing.unit * 2,
        paddingBottom: theme.spacing.unit * 2,

    }
});


function PaperSheet(props) {
    const { classes, tipo, disciplina, grupo, tempoRestante } = props;

    return (
        <div class="paper">
            <Paper 
                className={classes.root} 
                elevation={1}
                >
                <div class="left">
                    <Typography variant="h5" component="h3">
                        <b>{ tipo }</b> de <b>{ disciplina }</b> do <b>{ grupo }</b>
                    </Typography>
                </div>
                <div class="right">
                    <Typography component="h4">
                        Come√ßa em <b>{ tempoRestante} </b>
                    </Typography>
                </div>
            </Paper>
        </div>
    );
}

PaperSheet.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(PaperSheet);

