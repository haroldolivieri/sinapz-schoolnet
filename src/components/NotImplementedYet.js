import React from 'react';

const NotImplementedYet = () => (
    <div>
        <h1>Not implemented yet</h1>
    </div>
)

export default NotImplementedYet