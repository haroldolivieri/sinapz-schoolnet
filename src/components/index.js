export { default as NotFound } from './NotFound'
export * from './NotImplementedYet'
export { default as ResponsiveDrawer } from './ResponsiveDrawer'
export { default as SideDrawer } from './SideDrawer'
export { default as BranchList } from './BranchList'
export { default as SinapzCard } from './SinapzCard'