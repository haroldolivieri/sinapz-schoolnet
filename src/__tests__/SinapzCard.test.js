import React from 'react'
import { SinapzCard } from '../components'
import { createShallow } from '@material-ui/core/test-utils'
import CardHeader from '@material-ui/core/CardHeader'
import CardContent from '@material-ui/core/CardContent'
import Avatar from '@material-ui/core/Avatar'

describe(SinapzCard, () => {

  context('with branch Meier', () => {
    let shallow = createShallow({ dive: true })
    let branch = { id: 2, name: 'Meier', inCharge: 'Paula' }
    const wrapper = shallow(<SinapzCard branch={branch} />)
    
    it('should match with snapshot', () => { expect(wrapper).toMatchSnapshot() })
    it('should has CardHeader', () => { expect(wrapper.find(CardHeader)).toHaveLength(1) })
    it('should has CardContent with Meier text', () => {
      expect(wrapper.find(CardContent).find('.SinapzCard-title-2').children().text()).toEqual('Meier') 
    })
    it('should has CardContent with Paula text', () => {
      expect(wrapper.find(CardContent).find('.SinapzCard-inCharge-3').children().text()).toEqual('Paula') 
    })
    it('should has Avatar', () => { expect(wrapper.find(Avatar)).toBeDefined() })
  })
})
