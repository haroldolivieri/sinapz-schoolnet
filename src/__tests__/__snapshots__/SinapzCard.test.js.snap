// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithStyles with branch Meier should match with snapshot 1`] = `
<WithStyles(Card)
  className="SinapzCard-card-1"
  key="2"
>
  <WithStyles(Grid)
    container={true}
    direction="row"
    spacing={8}
  >
    <WithStyles(Grid)
      container={true}
      item={true}
      xs={1}
    >
      <WithStyles(CardHeader)
        avatar={
          <WithStyles(Avatar)>
            R
          </WithStyles(Avatar)>
        }
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={8}
    >
      <WithStyles(CardContent)>
        <WithStyles(Grid)
          item={true}
          xs={2}
        >
          <WithStyles(Typography)
            className="SinapzCard-title-2"
            gutterBottom={true}
            variant="h4"
          >
            Meier
          </WithStyles(Typography)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          xs={4}
        >
          <WithStyles(Typography)
            className="SinapzCard-inCharge-3"
            gutterBottom={true}
            variant="subtitle1"
          >
            Paula
          </WithStyles(Typography)>
        </WithStyles(Grid)>
      </WithStyles(CardContent)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Card)>
`;
